internal/controller/etcdcluster_controller.go

```diff
@@ -196,6 +200,52 @@ func (r *EtcdClusterReconciler) ensureClusterService(ctx context.Context, cluste
 	return nil
 }
 
+func (r *EtcdClusterReconciler) ensureClusterClientService(ctx context.Context, cluster *etcdaenixiov1alpha1.EtcdCluster) error {
+	svc := &corev1.Service{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: cluster.Namespace,
+		Name:      r.getClientServiceName(cluster),
+	}, svc)
+	// Service exists, skip creation
+	if err == nil {
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return fmt.Errorf("cannot get cluster client service: %w", err)
+	}
+
+	svc = &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.getClientServiceName(cluster),
+			Namespace: cluster.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":       "etcd",
+				"app.kubernetes.io/instance":   cluster.Name,
+				"app.kubernetes.io/managed-by": "etcd-operator",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{Name: "client", TargetPort: intstr.FromInt32(2379), Port: 2379, Protocol: corev1.ProtocolTCP},
+			},
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"app.kubernetes.io/name":       "etcd",
+				"app.kubernetes.io/instance":   cluster.Name,
+				"app.kubernetes.io/managed-by": "etcd-operator",
+			},
+			PublishNotReadyAddresses: false,
```

Comment:
probably we don't need to set `false` value for `PublishNotReadyAddresses` parameter explicitly , coz it's false by default.